### ISSUE TITLE
SyncService中线程池选用问题。

### ISSUE THREAD

**Issue by hero5512** [ontio-ontid/entity-trace#1]
使用newCachedThreadPool()线程池会不会导致内存溢出呢？

**Comment by leewi9** [comment]
我目前的想法是，如果为了防止内存溢出，可以在linux服务器上设置swap分区，但这个只是预防手段，可能没法解决根本问题。

不知道你有什么好的经验呢，请指教🤗 @hero5512

**Comment by hero5512** [comment]
> 我目前的想法是，如果为了防止内存溢出，可以在linux服务器上设置swap分区，但这个只是预防手段，可能没法解决根本问题。
> 
> 不知道你有什么好的经验呢，请指教🤗 @hero5512

我的想法是不使用Executors提供的线程池，而是使用ThreadPoolExecutor自己创建，同时限制缓存队列的大小，不知道这样可以么。

**Comment by hero5512** [comment]
还有个问题，这里如果节点挂掉这个循环会不会陷入死循环呢，建议设置一个备份节点切换的机制。
![图片 1](https://user-images.githubusercontent.com/10353921/55469207-9422a800-5637-11e9-8de5-f9b4ddb13221.png)

**Comment by leewi9** [comment]
@hero5512 是的。之前的我的设计是程序启动后，专门开一个线程去监测node是否挂掉，程序中所用的node设置成一个全局变量，程序中所有和链交互的方法都是从这个全局node发出，当监测进程监测到node挂掉，就用下一个url去初始化一个新的node给全局变量node，这样保证和链交互的方法始终用的都是有效node。

但这样，还是不完美，监测进程去试探node是否挂掉，是有一个interval的，比如这个间隔是10s去测试一次，那假如在10s刚开始node就挂掉，那这10s内和链交互的方法都会挂掉，直到下一次node切换。

不知道你有什么好的idea呢？
